src/version.rs

```diff
@@ -36,6 +38,24 @@ impl fmt::Display for HttpVersion {
     }
 }
 
+impl FromStr for HttpVersion {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<HttpVersion, Error> {
+        if s == "" {
+            Err(Error::Method)
```

Comment:
The error should probably be `Error::Version`.

Also, this if could be removed, and empty string would match the `_` arm in the match below.